src/history/hs_rec.c

```diff
@@ -686,6 +687,9 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
                 WT_STAT_CONN_DATA_INCR(session, cache_hs_write_squash);
                 squashed = false;
             }
+
+            if (upd == newest_hs)
```

Comment:
Even though you explained what the for loop is doing [line 512](https://github.com/wiredtiger/wiredtiger/pull/6984/commits/b09168b57501e42d2aa33361d599c9fd5a544d47#diff-43fd2cf13452e67cc179f7b306dc3d93b8db75cf5d0cf6d1cce293b1b260e434R512), which is great, we can add here that the work is finished as the newest HS value has been processed.